### ISSUE TITLE
pyright doesn't catch trivial unbount attributes in __init__

### ISSUE THREAD

**Issue by aubergenepizza** [microsoft/pyright#11384]
**Describe the bug**
Pyright doesn't detect unbound attributes in `__init__`. Strangely enough, other type-checkers don't catch this, too. Which means it might not be a bug?

**Code or Screenshots**

```python
class A:
    def __init__(self)->None:
        print(self.x)
        self.x: int = 2

a = A()
```

```bash
$ uvx pyright --version
pyright 1.1.408
$ uvx pyright a.py 
0 errors, 0 warnings, 0 informations
```
